compiler/bcq-tools/generate_bcq_metadata

```diff
@@ -0,0 +1,216 @@
+                break
+
+            if prefix not in prefix_infonames_dict:
+                prefix_infonames_dict[prefix] = set()
+
+            prefix_infonames_dict[prefix].add(infoname)
+
+    # All the number of BCQ information should be same
+    num_of_bcqinfo = -1
+    for key in prefix_infonames_dict:
+        infonames = prefix_infonames_dict[key]
+        if num_of_bcqinfo == -1:
+            num_of_bcqinfo = len(infonames)
+        elif num_of_bcqinfo != len(infonames):
+            is_valid = False
+
+    # The number of BCQv1 information should be 6 or 7
+    if num_of_bcqinfo != 6 and num_of_bcqinfo != 7:
+        is_valid = False
+
+    # If BCQ information is invalid, return original model
+    if is_valid == False:
+        return original_graph_def
+
+    new_graph_def = tf.compat.v1.GraphDef()
+    for node in original_graph_def.node:
+        new_node = new_graph_def.node.add()
+        new_node.CopyFrom(node)
+
+    # Generate metadata header
+    metadata_values = generate_metadata_header(original_graph, 1, flags.output_arrays)
+
+    # Append metadata of BCQv1
+    metadata_values = np.append(metadata_values, num_of_bcqinfo + 1)
+
+    # Finish generating metadata
+    metadata_values = np.append(metadata_values, ONE_END_MAGICNUM)
+
+    # Generate metadata tensor
+    metadata_tensor = tf.make_tensor_proto(metadata_values, tf.int32)
+
+    new_node = new_graph_def.node.add()
+    new_node.op = "Const"
+    new_node.name = "one_compiler/bcqinfo_one_metadata"
+    new_node.attr["dtype"].CopyFrom(
+        tf.core.framework.attr_value_pb2.AttrValue(type=tf.int32.as_datatype_enum))
+    new_node.attr["value"].tensor.CopyFrom(metadata_tensor)
+    return new_graph_def
+
+
+def determine_bcq_version(flags):
+    """
+    CAUTION : For now, BCQ has only one version and thus always returns 1 when BCQ
+    information nodes are included. If new BCQ version is introduced,
+    this function must be updated accordingly.
+
+    When BCQ information does not exist, -1 is returned.
+    """
+
+    bcq_version = -1
+
+    original_graph = load_graph(flags.input_path)
+    original_graph_def = original_graph.as_graph_def()
+
+    for node in original_graph_def.node:
+        if node.op == "Const" and "/bcqinfo_" in node.name:
+            bcq_version = 1
+            break
+
+    return bcq_version
+
+
+def generate_bcq_metadata(flags):
+    """
+    Basic format of metadata is as following.
+        - Magic number indicating start
+        - Version of BCQ Format
+        - The number of original outputs
+        - Metadata based on each BCQ format
+        - Magic number indicating end
+    """
+    version = determine_bcq_version(flags)
+
+    if version == 1:
+        result_graph_def = generate_bcq_metadata_v1(flags)
+    elif version == -1:
+        # When there is no BCQ information, do nothing
+        result_graph_def = load_graph(flags.input_path)
+    else:
+        raise Exception("BCQ version is not supported yet")
```

Comment:
One more thing :)
Can you show file version and this version number? Maybe like;
```
BCQ version of the file(number) is higher than the program(number). Please upgrade this program.
```